packages/astro/src/assets/build/generate.ts

```diff
@@ -207,6 +200,7 @@ export async function generateImagesForPath(
 			? (options.src as ImageMetadata).src
 			: (options.src as string);
 
+		const originalImage = await loadImage(originalFilePath, env);
```

Comment:
There's a tradeoff here with this that I made somewhat on purpose, this massively improve fully cached runs, especially with remote images, but it makes it way worse for initial builds with a lot of variants of images. Ideally we'd be able to come to a perfect solution, but I couldn't figure it out when I first did it though.

Reply:
Thanks for your input @Princesseuh! I’ll mark this PR as draft and try to investigate this scenario. Do you have any idea why it made it worse?

Reply:
Because you end up loading the image for every transform of an image. For instance, if you have an image of which you generate 16 variations (which is quite easy to do with Picture), you'll end up loading the image 17 times.

Probably that a refactor here would be possible where we check if every variants is cached, if so skip loading the image

Reply:
Got it, thanks!

Reply:
@Princesseuh, I've implemented your suggestions and updated the screenshots to include a non-cached run.